### PR TITLE
fix(generate daily fixed periods): use correct comparison for "endsBefore"

### PR DESCRIPTION
**Pull request by Mohammer5**
Title says everything ;)

No idea why there are no actions being run..
but `yarn build && yarn lint && yarn cucumber && yarn test --watchAll=false` exits with `0`, so everything should be fine

### CODE REVIEW

File: src/period-calculation/generate-fixed-periods/generate-fixed-periods-daily.ts
Comment (tomzemp) [inline src/period-calculation/generate-fixed-periods/generate-fixed-periods-daily.ts:37]: it might be a naming issue, but in isolation, I would think that this should be `Temporal.PlainDate.compare(nextDay, endBefore) > 0` ? Like now it is not breaking if nextDay and endBefore are the same...but by the name endBefore, I would think nextDay has to be before endBefore?

**Review by tomzemp — APPROVED** [review]
I approve, but not sure if you want to look at the comments from me and Mozafar before merging

File: src/period-calculation/generate-fixed-periods/generate-fixed-periods-daily.ts
Comment (Mohammer5) [reply src/period-calculation/generate-fixed-periods/generate-fixed-periods-daily.ts:35]: The `nextDay` date is calculated with `.add({ days: i })`. Do you suggest to convert that back into a string?

File: src/period-calculation/generate-fixed-periods/generate-fixed-periods-daily.ts
Comment (Mohammer5) [reply src/period-calculation/generate-fixed-periods/generate-fixed-periods-daily.ts:37]: > but by the name endBefore, I would think nextDay has to be before endBefore?

Yes, that is the case now. When
* nextDay < endsBefore, then `compare` will yield `-1`
* nextDay = endsBefore, then `compare` will yield `0`
* nextDay > endsBefore, then `compare` will yield `1`

So with the current implementation, it works as expected (confirmed by the test)

**Pull request merged**
